tooling/greybox_fuzzer/src/mutation/string.rs

```diff
@@ -0,0 +1,290 @@
+use crate::mutation::configurations::{
+    TopLevelMutation, BASIC_SPLICE_MUTATION_CONFIGURATION, BASIC_TOP_LEVEL_MUTATION_CONFIGURATION,
+};
+
+use super::configurations::{
+    ByteValueMutation, ByteValueMutationConfiguration, SpliceCandidate, SpliceMutation,
+    StructuralMutation, BASIC_BYTE_VALUE_MUTATION_CONFIGURATION,
+    BASIC_SPLICE_CANDIDATE_PRIORITIZATION_CONFIGURATION, BASIC_STRUCTURE_MUTATION_CONFIGURATION,
+    DICTIONARY_EMPTY_BYTE_VALUE_MUTATION_CONFIGURATION,
+};
+use super::dictionary::IntDictionary;
+use acvm::{AcirField, FieldElement};
+use noirc_abi::input_parser::InputValue;
+use rand::{seq::SliceRandom, Rng};
+use rand_xorshift::XorShiftRng;
+use std::cmp::min;
+
+/// This file contains mechanisms for mutating string InputValues. It can perform the following mutations:
+/// 1. Value mutations
+///     a. Substitution a random character in the string with a random appropriate value from the dictionary
+///     b. Substitution of a random character with a random character
+/// 2. Structural mutations
+///     a. Chaotically splicing the string with itself (constructing a new string from random chunks of initial string)
+///     b. Duplication of a random chunk (picking a chunk of the string and inserting it several times)
+///     c. Inserting a repeated random value
+///     d. Swapping two chunks
+///
+/// It also contains the splicing mechanism used when splicing two inputs. It chooses between:
+/// 1. Structured splicing (preserving the indices of the values)
+/// 2. Chaotic (creating the new string from chunks of the two given strings)
+///
+const MIN_ASCII: u8 = 0x0;
+const MAX_ASCII: u8 = 0x7f;
+
+struct StringMutator<'a> {
+    dictionary: &'a Vec<FieldElement>,
+    prng: &'a mut XorShiftRng,
+    value_mutation_configuration: &'static ByteValueMutationConfiguration,
+}
+
+impl<'a> StringMutator<'a> {
+    pub fn new(dictionary: &'a IntDictionary, prng: &'a mut XorShiftRng) -> Self {
+        let u8_dictionary = dictionary.get_dictionary_by_width(8);
```

Comment:
Would it make sense for the dictionary here to only consist of bytes that are ascii?